File: golang/330.patching-array.go
func minPatches(nums []int, n int) int {
	cover := 0
	i := 0
	patches := 0

	for cover < n {
		if i == len(nums) || nums[i] > cover+1 {
			cover += cover + 1
			patches++
		} else {
			cover += nums[i]
			i++
		}
	}

	return patches
}

File: cpp/330.patching-array.cpp
#include <vector>

using std::vector;

class Solution {
public:
  int minPatches(vector<int> &nums, int n) {
    int patches = 0, i = 0;
    long cover = 0;

    while (cover < n) {
      if (i < nums.size() && nums[i] <= cover + 1) {
        cover += nums[i++];
      } else {
        cover += cover + 1;
        patches++;
      }
    }

    return patches;
  }
};